src/gsi/setupfed.f90

```diff
@@ -548,24 +562,31 @@ subroutine setupfed(obsLL,odiagLL,lunin,mype,bwork,awork,nele,nobs,is,fed_diagsa
      !--------------Calculate departure from observation----------------!
      
      ddiff = data(ifedob,i) - FEDMdiag(i)
-
-!    If requested, setup for single obs test.
-!    Note: do not use this way to run single obs test for fed in the current version. (g.zhao)
-     if (oneobtest) then
-        ddiff=maginnov
-!        if (trim(adjustl(oneob_type))=='fed') then
-!           data(ifedob,i) = maginnov
-!           ddiff = data(ifedob,i) - FEDMdiag(i)
-!        end if
+     if (dofedoneob) then
```

Comment:
when do we need a new dofedoneob in addition to oneobtest?

Reply:
@TingLei-NOAA 
This is similar to the dBZ DA, which added more variables for one obs tests. 
I think this is because the dBZ and FED obs are in NetCDF format. If using the normal ONE OB test, it will triger GSI to write out a bufr file of single obs. This doesn't work because there is no Bufr table for dBZ and FED obs.

Reply:
OK, this makes sense!